src/java/voldemort/client/protocol/admin/AdminClient.java

```diff
@@ -1875,10 +1877,16 @@ public void verifyOrAddStore(StoreDefinition newStoreDef, String localProcessNam
                 try {
                     // Get all StoreDefinitions from each nodes in the cluster
                     remoteStoreDefs = metadataMgmtOps.getRemoteStoreDefList(nodeId).getValue();
-                } catch (UnreachableStoreException e) {
-                    logger.warn("Failed to contact " + node.briefToString() + " in order to validate the StoreDefinition.");
-                    unreachableNodes.add(node);
-                    continue;
+                } catch (Exception e) {
+                    if (ExceptionUtils.recursiveClassEquals(e,
+                                                            UnreachableStoreException.class,
+                                                            IOException.class)) {
+                        logger.warn("Failed to contact " + node.briefToString() + " in order to validate the StoreDefinition.");
+                        unreachableNodes.add(node);
+                        continue;
+                    } else {
+                        throw new VoldemortException("Unhandled exception during verifyOrAddStore() !", e);
```

Comment:
This function ```throws UnreachableStoreException, VoldemortException``` so it does not allow throwing the untyped ```Exception e```. Therefore, wrapping whatever ```Exception``` in ```VoldemortException``` is not a semantic change, I think.